### ISSUE TITLE
The div_factor parameter in OneCycleLR does not work

### ISSUE THREAD

**Issue by timothyman** [pytorch/pytorch#28216]
## 🐛 Bug

The `div_factor ` parameter determines the initial learning rate via `initial_lr = max_lr/div_factor` in the OneCycleLR scheduler.  The default is 25, but this parameter does nothing.

## To Reproduce

1. Create a OneCycleLR scheduler
```python
model = models.resnet18(pretrained=False)
model.train()
optimizer = Adam(model.parameters(), lr=1E-2)
lr_scheduler = OneCycleLR(optimizer, 
                          max_lr=1E-2, 
                          total_steps=100, 
                          cycle_momentum=False,
                          div_factor=25)
```
2. Plot the Learning Rate
```python
lr = []
for epoch in range(100):
    optimizer.zero_grad()
    optimizer.step()
    lr_scheduler.step()
    for param_group in optimizer.param_groups:
        lr.append(param_group['lr'])
plt.plot(lr);
```
3. Change the `div_factor` and plot again (step 2).
```python
lr_scheduler = OneCycleLR(optimizer, 
                          max_lr=1E-2, 
                          total_steps=100, 
                          cycle_momentum=False,
                          div_factor=10000)
```

## Expected behavior

The initial learning rate should change as a function of `div_factor`.

## Environment

 - PyTorch Version (e.g., 1.0): 1.3.0
 - OS (e.g., Linux): Windows
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

cc @vincentqb

**Comment by smessmer** [comment]
Thanks for reporting this. We'll take a look at your PR.

**Comment by timothyman** [comment]
While writing a unit-test, I found that the issue is not as straight-forward as I was initially led to believe.

This test fails in PyTorch 1.3.0.

 ```python
def test_onecycle_lr_div_factors(self):
        # https://github.com/pytorch/pytorch/issues/28216

        model = torch.nn.Linear(2, 1)
        optimizer = torch.optim.SGD(model.parameters(), lr=0.5)
        div_factors = [1, 10, 25, 100]
        final_div_factors = [1, 1E2, 1E4, 1E6]

        for div_factor, final_div_factor in zip(div_factors, final_div_factors):
            for strategy in ['linear', 'cos']:
                scheduler = OneCycleLR(optimizer, max_lr=25, total_steps=10,
                                       anneal_strategy=strategy,
                                       cycle_momentum=False,
                                       div_factor=div_factor,
                                       final_div_factor=final_div_factor
                                       )
                lrs = []
                for i in range(10):
                    lrs.append(optimizer.param_groups[0]["lr"])
                    optimizer.step()
                    scheduler.step()

                initial_lr, max_lr, last_lr = lrs[0], max(lrs), lrs[-1]
                self.assertAlmostEqual(max_lr / initial_lr, div_factor)
                self.assertAlmostEqual(initial_lr, final_div_factor * last_lr)
```
This alternate version, however, passes:

```python
def test_onecycle_lr_div_factors(self):
        # https://github.com/pytorch/pytorch/issues/28216

        model = torch.nn.Linear(2, 1)
        div_factors = [1, 10, 25, 100]
        final_div_factors = [1, 1E2, 1E4, 1E6]

        for div_factor, final_div_factor in zip(div_factors, final_div_factors):
            for strategy in ['linear', 'cos']:
                optimizer = torch.optim.SGD(model.parameters(), lr=0.5)
                scheduler = OneCycleLR(optimizer, max_lr=25, total_steps=10,
                                       anneal_strategy=strategy,
                                       cycle_momentum=False,
                                       div_factor=div_factor,
                                       final_div_factor=final_div_factor
                                       )
                lrs = []
                for i in range(10):
                    lrs.append(optimizer.param_groups[0]["lr"])
                    optimizer.step()
                    scheduler.step()

                initial_lr, max_lr, last_lr = lrs[0], max(lrs), lrs[-1]
                self.assertAlmostEqual(max_lr / initial_lr, div_factor)
                self.assertAlmostEqual(initial_lr, final_div_factor * last_lr)
```
 The only difference is whether I instantiate a new optimizer version for every new LR scheduler. It looks like the optimizer keeps the state of the previous LR schedule. I'm not sure if this is intended behavior?

Both versions pass in my pull request #28217

**Comment by vincentqb** [comment]
Schedulers are now chainable. The first case is simply applying the new scheduler where the other left off. You do need to create a new optimizer. If both are passing, the scheduler is therefore not chainable in its implementation. If there's a way to preserve that property, we should.
